volumina/utility/preferences.py

```diff
@@ -0,0 +1,143 @@
+###############################################################################
+#   volumina: volume slicing and editing library
+#
+#       Copyright (C) 2011-2014, the ilastik developers
+#                                <team@ilastik.org>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the Lesser GNU General Public License
+# as published by the Free Software Foundation; either version 2.1
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# See the files LICENSE.lgpl2 and LICENSE.lgpl3 for full text of the
+# GNU Lesser General Public License version 2.1 and 3 respectively.
+# This information is also available on the ilastik web site at:
+# 		   http://ilastik.org/license/
+###############################################################################
+
+"""Application-wide persistent preferences.
+
+All public functions in this module are thread-safe.
+"""
+
+import logging
+import os
+import pathlib
+import pickle
+import threading
+from typing import Any, Tuple, MutableMapping
+
+logger = logging.getLogger(__name__)
+
+
+class _Preferences:
+    def __init__(self, path: pathlib.Path):
+        self._path = path
+        self._data = _load_preferences(path)
+        self._lock = threading.Lock()
+
+    def get(self, group: str, setting: str, default: Any = None) -> Any:
+        return self.getmany((group, setting, default))[0]
```

Comment:
Do we really need getmany?

Reply:
It is not used anywhere, but was added for the symmetry with `setmany`.